distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/JvmOptionsParser.java

```diff
@@ -90,8 +91,10 @@ public void accept(final int lineNumber, final String line) {
             final List<String> substitutedJvmOptions =
                 substitutePlaceholders(jvmOptions, Map.of("ES_TMPDIR", System.getenv("ES_TMPDIR")));
             final List<String> ergonomicJvmOptions = JvmErgonomics.choose(substitutedJvmOptions);
-            substitutedJvmOptions.addAll(ergonomicJvmOptions);
-            final String spaceDelimitedJvmOptions = spaceDelimitJvmOptions(substitutedJvmOptions);
+            final List<String> finalJvmOptions = Stream.concat(
```

Comment:
I wondered if this would be any better using a plain old list:

```java
final List<String> finalJvmOptions = new ArrayList<>(SystemJvmOptions.systemJvmOptions());
finalJvmOptions.addAll(substitutedJvmOptions);
finalJvmOptions.addAll(ergonomicJvmOptions);